src/modm/board/disco_f411ve/board.hpp

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2024, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+#pragma once
+
+#include <modm/architecture/interface/clock.hpp>
+#include <modm/driver/inertial/lis3dsh.hpp>
+#include <modm/platform.hpp>
+
+using namespace modm::platform;
+
+namespace Board
+{
+/// @ingroup modm_board_disco_f411ve
+/// @{
+using namespace modm::literals;
+
+/// STM32F411 running at 100MHz generated from the external 8MHz crystal
+struct SystemClock
+{
+	static constexpr uint32_t Frequency = 100_MHz;
+	static constexpr uint32_t Ahb = Frequency;
+	static constexpr uint32_t Apb1 = Frequency / 2;
+	static constexpr uint32_t Apb2 = Frequency;
+
+	static constexpr uint32_t Adc = Apb2;
+
+	static constexpr uint32_t Spi1 = Apb2;
+	static constexpr uint32_t Spi2 = Apb1;
+	static constexpr uint32_t Spi3 = Apb1;
+	static constexpr uint32_t Spi4 = Apb2;
+	static constexpr uint32_t Spi5 = Apb2;
+
+	static constexpr uint32_t Usart1 = Apb2;
+	static constexpr uint32_t Usart2 = Apb1;
+	static constexpr uint32_t Usart3 = Apb1;
+	static constexpr uint32_t Uart4 = Apb1;
+	static constexpr uint32_t Uart5 = Apb1;
+	static constexpr uint32_t Usart6 = Apb2;
+	static constexpr uint32_t Uart7 = Apb1;
+	static constexpr uint32_t Uart8 = Apb1;
+
+	static constexpr uint32_t I2c1 = Apb1;
+	static constexpr uint32_t I2c2 = Apb1;
+	static constexpr uint32_t I2c3 = Apb1;
+
+	static constexpr uint32_t Apb1Timer = Apb1 * 2;
+	static constexpr uint32_t Apb2Timer = Apb2 * 2;
+	static constexpr uint32_t Timer1 = Apb2Timer;
+	static constexpr uint32_t Timer2 = Apb1Timer;
+	static constexpr uint32_t Timer3 = Apb1Timer;
+	static constexpr uint32_t Timer4 = Apb1Timer;
+	static constexpr uint32_t Timer5 = Apb1Timer;
+	static constexpr uint32_t Timer9 = Apb2Timer;
+	static constexpr uint32_t Timer10 = Apb2Timer;
+	static constexpr uint32_t Timer11 = Apb2Timer;
+
+	static constexpr uint32_t Usb = 48_MHz;
```

Comment:
Is this correct?

Reply:
I've setup a PWM on Timer10 and Timer11, confirmed the output frequency and looks fine.

But could not make USB / TinyUSB working.
Please give me some advice!

Observations:
- Power on without Usb-Cable, Leds blink period = 2.5s -> Seems to be the initial default
- Connecting an Usb-Cable, the Leds blink period remains at 2.5s -> tud_mount_cb() has not been iunvoked.
- However, when disconnecting the Usb-Cable, the blink period changes to 250ms -> tud_umount_cb() has been iunvoked!
- Reconnecting the cable again, changes nothing. The blink period remains at 250ms.
- Only a reset gets me back to 2.5s right now.

Reply:
That's cos the USB is running at 50MHz and not 48MHz, hence my question about whether this clock is correct ;-P It probably works for a short time, but then gets out of sync.

CubeMX shows this solution running the CPU at 96MHz and USB at 48MHz:
![](https://github.com/modm-io/modm/assets/163066/e4233797-2dbe-45b2-aa7f-2a9c4ef9c53c)

Reply:
Actually, the /Q prescaler isn't set at all, so the USB probably runs at 100MHz or something completely wrong.

Reply:
Ok, i have not really tried to iunderstand the clock system yet. So i got a copy of CubeMX as it looks like an usefull reference. Have seen this graphical configurator before but your screenshot finally convinced me to give it a closer look, thanks!

USB now works 😄

Reply:
I swapped the prescalers to align with the CubeMX solution, so the CPU now runs at 96MHz and the USB still at 48MHz.